### PR TITLE
Display hearing time timezones on schedule veterans page

### LINKED ISSUE #15394: Fix hearing timezone when scheduling a virtual to a video docket
#### Summary
Hearing Time isn't displaying in RO time when scheduling a virtual hearing to a video docket day

#### Steps to reproduce
1. Complete steps 1-5 from Scenario A with https://appeals.cf.uat.ds.va.gov/queue/appeals/1063943
1. Select the Other radio button under Hearing Time, then choose any option from the dropdown [chose **2pm Eastern Time**]
1. Navigate to /hearings/schedule
1. On Daily Docket, find the date (11/17), RO (St Louis), and room that you selected from the Hearing Date dropdown

#### Actual behavior
- On schedule page, chose Hearing Time of **2pm Eastern Time**
- On Daily Docket, saw that the Hearing Time displays as **3pm Eastern Time / 2pm Central Time**
- In veteran and representative emails, see that the Hearing Time displays as **2pm Central Time**

#### Desired behavior
The Hearing Time dropdown on the Schedule page displays Regional Office (RO) time when a video docket day is selected. In this case, that would be CDT for St. Louis, as is displayed on the Daily Docket.

##### Schedule page - hearing time in Eastern time

<hr>

##### Daily Docket - hearing time in RO time (Central time)

<hr>

### PR DESCRIPTION
**Pull request by ferristseng**
Resolves #15394 

### Description

 - Updates the `RegionalOffice` dropdown options to include the entire regional office object instead of just the RO key
   - Lots of small updates related to this for other components that use the `RegionalOffice` component
 - Pass in `localTimezone` to `HearingTime` components
 - Update tests and storybook

### Test Plan
1. Shadow user `BVASYELLOW`
1. Navigate to the schedule hearings page http://localhost:3000/hearings/schedule/assign
1. Choose the "St. Petersburg, FL" regional office and choose the schedule veterans tab and click on one of the appeals to schedule a veteran
1. On the case details page, click the "Schedule Veteran" action from the dropdown
1. Ensure that the regional office is populated with the RO you chose on the assign hearings page
1. Finish scheduling the veteran and once complete go through the postpone workflow from the case details page to ensure that the RO is still selected

### User Facing Changes

---

Screen Shot 2020-10-07 at 5 40 49 PM

---

Screen Shot 2020-10-07 at 5 40 45 PM

### CODE REVIEW

File: client/app/components/DataDropdowns/RegionalOffice.jsx [hunk 1]
```diff
@@ -42,7 +41,7 @@ class RegionalOfficeDropdown extends React.Component {
       _.forEach(resp.regionalOffices, (value, key) => {
         regionalOfficeOptions.push({
           label: value.state === 'DC' ? 'Central' : `${value.city}, ${value.state}`,
-          value: key
```
Comment (ferristseng) [inline client/app/components/DataDropdowns/RegionalOffice.jsx:45]: Updates the `RegionalOffice` options to preserve the entire regional office instance to make the timezone is available

File: client/app/hearings/components/ScheduleVeteran.jsx [hunk 2]
```diff
@@ -80,6 +80,8 @@ export const ScheduleVeteran = ({
   // Create a hearing object for the form
   const hearing = {
     ...assignHearingForm,
+    regionalOffice: assignHearingForm?.regionalOffice?.key,
+    regionalOfficeTimezone: assignHearingForm?.regionalOffice?.timezone,
```
Comment (ferristseng) [inline client/app/hearings/components/ScheduleVeteran.jsx:84]: Populate the `regionalOffice` and `regionalOfficeTimezone` fields which is used in `ScheduleVeteranForm`

File: client/app/hearings/components/ScheduleVeteranForm.stories.js [hunk 3]
```diff
@@ -66,20 +66,17 @@ CentralToVirtualConversion.args = {
...
-  },
+    regionalOffice: centralHearing.regionalOfficeKey,
+  }
 };
 
 export const VideoWithErrors = Template.bind({});
 VideoWithErrors.args = {
   hearing: {
     ...defaultHearing,
-    requestType: defaultHearing.readableRequestType,
```
Comment (ferristseng) [inline client/app/hearings/components/ScheduleVeteranForm.stories.js:77]: `requestType` is no longer passed in with the `hearing` in `ScheduleVeteran`

File: client/test/data/regional-offices.js [hunk 4]
```diff
@@ -6,233 +9,10 @@ export const regionalOfficeCities = {
...
-    value: 'RO52'
-  },
-  {
-    label: 'Winston-Salem, NC',
-    value: 'RO18'
-  }
-]
-;
+export const roList = map(
+  regionalOfficesJsonResponse['regional_offices'],
```
Comment (ferristseng) [inline client/test/data/regional-offices.js:13]: I saved the response from the API and am using it to generate the list of RO options for the dropdown

**Review by sahalliburton — CHANGES_REQUESTED** [review]
Overall LGTM, but when you try postponing the hearing, it no longer preselects the regional office

**Review by tomas-nava — APPROVED** [review]
approve!

(Stephen & I approved each others' changes)

**Review by sahalliburton — APPROVED** [review]
LGTM!

**Pull request merged**